packages/babel-helpers/src/helpers.js

```diff
@@ -372,6 +372,25 @@ helpers.extends = defineHelper(`
   }
 `);
 
+helpers.spreadByDefine = defineHelper(`
+  export default function _spreadByDefine(target) {
+    for (var i = 1; i < arguments.length; i++) {
+      var source = arguments[i];
+      for (var key in source) {
```

Comment:
`for in` does not enumerate `Symbol` keys, but they should be copied when they are enumerable.

`ForIn / OfHeadEvaluation` will call this when evaluating a `for in`: https://tc39.github.io/ecma262/#sec-enumerate-object-properties

`CopyDataProperties` calls `[[OwnPropertyKeys]]`, though, which almost always delegates to https://tc39.github.io/ecma262/#sec-ordinaryownpropertykeys, which includes symbols.

(🚲🏠  maybe `copyDataProperties` is a better name for the helper 🤔)

Reply:
It also seems that `Reflect.ownKeys` does what we want. Since it's now possible for helpers to depend on other helpers, maybe a `ownPropertyKeys` helper can be made.

```js
export default function _ownPropertyKeys (target) {
  if (typeof Reflect !== "undefined" && Reflect.ownKeys) {
    _ownPropertyKeys = Reflect.ownKeys;
  } else {
    _ownPropertyKeys = ownKeysImpl;
  }
  return _ownPropertyKeys.apply(this, arguments);
  function ownKeysImpl (target) {
    const keys = Object.keys(target)
    if (Object.getOwnPropertySymbols) {
      return keys.concat(
        Object.getOwnPropertySymbols(target)
          .filter(sym => Object.getOwnPropertyDescriptor(target, sym).enumerable)
      );
    }
    return keys;
  }
}
```

Reply:
> for in does not enumerate Symbol keys, but they should be copied when they are enumerable.

:man_facepalming: Oh man... Indeed, `for in` isn't ok in this context. Moreover, it enumerates properties in the prototype chain as well, so in any case `Object.keys()` should be used... Thanks for the additional review! I need to add a few more tests… 

> It also seems that Reflect.ownKeys does what we want. Since it's now possible for helpers to depend on other helpers, maybe a ownPropertyKeys helper can be made.

👍  Yeah, that could be a great solution! The only thing I see is that in `ownKeysImpl ` you're additionally filtering for `enumerable` symbols, when in reality `Reflect.ownKeys` is basically the same as `Object.getOwnPropertyNames(target).concat(Object.getOwnPropertySymbols(target))`… so I guess if we add an `_ownPropertyKeys ` helper it needs to return the same as `Reflect.ownKeys`... Additionally, it would just be an internal helper of an helper in this context 😄  so given your example, I'd say that this would be enough for now:

```js
function spreadByDefine(target) {
  for (var i = 1; i < arguments.length; i++) {
    var source = arguments[i];
    var ownKeys = Object.keys(source);
    if (typeof Object.getOwnPropertySymbols === 'function') {
      ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) { 
        return Object.getOwnPropertyDescriptor(source, sym).enumerable;
      }));
    }
    ownKeys.forEach(function(key) {
      Object.defineProperty(target, key, {
        value: source[key],
        writable: true,
        enumerable: true,
        configurable: true
      });
    });
  }
  return target;
}
```

> maybe copyDataProperties is a better name for the helper

TBH, this helper is doing almost exactly the same as the `extends` one, so I'd say that `extendsByDefine` would be a good name (originally was `extendsDefine`, now `spreadByDefine`…) 💭